controllers/medusa/medusabackupjob_controller_test.go

```diff
@@ -263,21 +266,36 @@ func createAndVerifyMedusaBackup(dcKey framework.ClusterKey, dc *cassdcapi.Cassa
 		verifyBackupJobStarted(require.Never, t, dc, f, ctx, backupKey)
 	}
 
-	// TODO That previous check does not actually verify that the MedusaBackup has yet been written (if it should be), so we need to wait a bit more
-	time.Sleep(500 * time.Millisecond)
-
-	t.Log("check for the MedusaBackup being created")
-	medusaBackupKey := framework.NewClusterKey(dcKey.K8sContext, dcKey.Namespace, backupName)
 	medusaBackup := &api.MedusaBackup{}
-	t.Logf("getting MedusaBackup %s", medusaBackupKey)
-	err = f.Get(ctx, medusaBackupKey, medusaBackup)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			// exit the test if the backup was not created. this happens for some backup names on purpose
-			return false
-		}
+	if backupName != backupWithNoPods && strings.HasPrefix(backupName, "good") {
+		require.Eventually(func() bool {
+			// medusaBackup := &api.MedusaBackup{}
+			if err := f.Get(ctx, backupKey, medusaBackup); err != nil {
+				t.Logf("failed to get MedusaBackup: %v", err)
+				if errors.IsNotFound(err) {
+					// exit the test if the backup was not created. this happens for some backup names on purpose
+					return false
+				}
+			}
+			return true
+		}, timeout, interval)
+	} else {
+		return false
 	}
 
+	/*
```

Comment:
Unless there's a good reason leaving this commented, we should either uncomment or remove.

Reply:
I can remove it.